src/SigningPortal/SigningPortal.tsx

```diff
@@ -62,11 +65,28 @@ export const SigningPortal: React.FC = () => {
 
   const { selectedAccount } = useAccounts();
 
-  // Check if the transaction is related to the Registrar pallet: reserve a parachain id or register a parachain.
+  // Helper function to check if a transaction is a Registrar reserve/register call
+  const isRegistrarCall = (call: any): boolean => {
+    return call?.type === "Registrar" && ["reserve", "register"].includes(call?.value?.type);
+  };
+
+  // Helper function to check if a wrapped call (Proxy or Sudo) is a Registrar call
+  const isWrappedRegistrarCall = (type: string, value: any): boolean => {
+    return (
+      (type === "Proxy" && value?.type === "proxy" && isRegistrarCall(value?.value?.call)) ||
```

Comment:
```suggestion
      (type === PROXY_PALLET && value?.type === PROXY_PALLET_CALL && isRegistrarCall(value?.value?.call)) ||
```
A constant for these strings would make code easier to maintain. If you agree, please apply for `Registrar` and `Sudo` as well